tfjs-core/src/ops/browser.ts

```diff
@@ -358,7 +394,7 @@ export async function toPixels(
     bytes[j + 3] = Math.round(rgba[3]);
   }
 
-  if (canvas != null) {
+  if (canvas != null && kernel == null) {
     canvas.width = width;
     canvas.height = height;
     const ctx = canvas.getContext('2d');
```

Comment:
@pyu10055 @mattsoulanille It's possible that `canvas.getContext('2d')` is null if a `webgl` or `webgpu`  context has been requested for the same canvas before the canvas passes into `toPixels`. Here we should check `ctx`'s value. And what's this API's expectation for `canvas`? A canvas which is never called `getContext` or a `2d` canvas?

Reply:
@Linchenn, I think you might have more context on this.

Reply:
I add a check as below:
+  if (!gpuContext) {
+    throw new Error(
+        `Please make sure this canvas has only been used for webgpu context!`);
+  }

Reply:
We may not need to expect the canvas and context.

If WebGL backend could draw to the canvas (same canvas as the backend and uses the same context), then draw it directly; otherwise, use CPU way to draw to it, but we could send a warning for this.